### PR TITLE
[MPS] Fix convolution `Source and weight input channels mismatch' crash

### PR DESCRIPTION
**Pull request by DenisVieriu97**
Fixes crashes in conv input/weight backward passes due to NCHW / NHWC formats.

### CODE REVIEW

**Review by razarmehr — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/mps/operations/Convolution.mm [hunk 1]
```diff
@@ -39,6 +39,19 @@ void fill_conv_desc(MPSGraphConvolution2DOpDescriptor* descriptor_,
   descriptor_.groups = groups;
 }
 
+static
+MPSShape* get_mps_conv_shape(const Tensor& tensor, bool is_channels_last) {
```
[the remark this answers is not in the document]
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/Convolution.mm:43]: I'll fix that - I've picked up directly what was available on kulin's master during conflicts. Thanks

File: aten/src/ATen/native/mps/operations/Convolution.mm [hunk 2]
```diff
@@ -126,7 +139,7 @@ Tensor _mps_convolution(
                                     + mps::getTensorsStringKey({input_t, weight_t}) + ":"
                                     + to_string(bias_defined) + ":" + bias_shape_key;
     CachedGraph* cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
-    MPSShape* inputShape = mps::getMPSShape(input_t, memory_format);
+    MPSShape* inputShape = get_mps_conv_shape(input_t, is_channels_last);
```
[the remark this answers is not in the document]
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/Convolution.mm]: Thanks. Updated

File: aten/src/ATen/native/mps/operations/Convolution.mm [hunk 3]
```diff
@@ -311,14 +340,18 @@ Tensor mps_convolution_backward_input(
...
 Tensor mps_convolution_backward_weights(
-    IntArrayRef weight_size, const Tensor& grad_output_t, const Tensor& input_t,
+    IntArrayRef weight_size, const Tensor& grad_output_, const Tensor& input_,
     IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups, bool bias_defined) {
   namespace native_mps = at::native::mps;
   using namespace mps;
   CheckedFrom c = "mps_convolution_backward_weights";
-  auto memory_format = input_t.suggest_memory_format();
-  MPSShape* inputShape = getMPSShape(input_t, memory_format);
-  MPSShape* gradOutputShape = getMPSShape(grad_output_t, memory_format);
```
[the remark this answers is not in the document]
Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/Convolution.mm:321]: Thanks, updated

**Review by razarmehr — APPROVED** [review]
LGTM

**Pull request closed without merging**